pkg/cmd/ssh/command_test.go

```diff
@@ -0,0 +1,481 @@
+/*
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ssh
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	internalfake "github.com/gardener/gardenctl-v2/internal/fake"
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/config"
+	"github.com/gardener/gardenctl-v2/pkg/target"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	operationsv1alpha1 "github.com/gardener/gardener/pkg/apis/operations/v1alpha1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func init() {
+	utilruntime.Must(gardencorev1beta1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(operationsv1alpha1.AddToScheme(scheme.Scheme))
+}
+
+type bastionStatusPatch func(status *operationsv1alpha1.BastionStatus)
+
+func waitForBastionThenPatchStatus(ctx context.Context, gardenClient client.Client, bastionName string, namespace string, patcher bastionStatusPatch) {
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+
+		case <-ticker.C:
+			key := types.NamespacedName{Name: bastionName, Namespace: namespace}
+			bastion := &operationsv1alpha1.Bastion{}
+
+			if err := gardenClient.Get(ctx, key, bastion); err != nil {
+				break
+			}
+
+			patch := client.MergeFrom(bastion.DeepCopy())
+			patcher(&bastion.Status)
+
+			Expect(gardenClient.Status().Patch(ctx, bastion, patch)).To(Succeed())
+
+			return
+		}
+	}
+}
```

Comment:
```suggestion
func waitForBastionThenPatchStatus(ctx context.Context, gardenClient client.Client, bastionName string, namespace string, patcher bastionStatusPatch) {
	defer GinkgoRecover()

	ticker := time.NewTicker(1 * time.Second)
	defer ticker.Stop()

	for {
		select {
		case <-ctx.Done():
			return

		case <-ticker.C:
			key := types.NamespacedName{Name: bastionName, Namespace: namespace}

			Eventually(func() error {
				bastion := &operationsv1alpha1.Bastion{}

				if err := gardenClient.Get(ctx, key, bastion); err != nil {
					return err
				}

				patch := client.MergeFrom(bastion.DeepCopy())
				patcher(&bastion.Status)

				return gardenClient.Status().Patch(ctx, bastion, patch)
			}).Should(Succeed())

			return
		}
	}
}
```

We have to add `defer GinkgoRecover()` as this function is always run in a go routine.

I also had a case where the status patch failed with an error that the bastion was modified. That's why I wrapped it with `Eventually`